synthetics/provider.go

```diff
@@ -60,28 +44,147 @@ func NewProvider() *schema.Provider {
 	}
 }
 
+func providerSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		apiURLKey: {
+			Type:        schema.TypeString,
+			Optional:    true,
+			DefaultFunc: schema.EnvDefaultFunc("KTAPI_URL", nil),
+			Description: "Synthetics API server URL (optional). Can also be specified with KTAPI_URL environment variable.",
+		},
+		emailKey: {
+			Type:        schema.TypeString,
+			Required:    true,
+			DefaultFunc: schema.EnvDefaultFunc("KTAPI_AUTH_EMAIL", nil),
+			Description: "Authorization email (required). Can also be specified with KTAPI_AUTH_EMAIL environment variable.",
+		},
+		tokenKey: {
+			Type:        schema.TypeString,
+			Sensitive:   true,
+			Required:    true,
+			DefaultFunc: schema.EnvDefaultFunc("KTAPI_AUTH_TOKEN", nil),
+			Description: "Authorization token (required). Can also be specified with KTAPI_AUTH_TOKEN environment variable.",
+		},
+		retryKey: {
+			Type:        schema.TypeList,
+			Optional:    true,
+			MaxItems:    1,
+			ForceNew:    true,
+			Description: "Configuration for API client retry mechanism",
+			Elem: &schema.Resource{
+				Schema: retryProperties(),
+			},
+		},
+		logPayloadsKey: {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			DefaultFunc: schema.EnvDefaultFunc("KTAPI_LOG_PAYLOADS", false),
+			Description: "Log payloads flag enables verbose debug logs of requests and responses (optional). " +
+				"Can also be specified with KTAPI_LOG_PAYLOADS environment variable.",
+		},
+	}
+}
+
+// retryProperties groups retry properties.
+// Note that, default values of schema.EnvDefaultFunc are not used, because they are not applied when user does not pass
+// retry block at all.
+func retryProperties() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		maxAttemptsKey: {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			DefaultFunc: schema.EnvDefaultFunc("KTAPI_RETRY_MAX_ATTEMPTS", nil),
+			Description: "Maximum number of request retry attempts. Default: 100. " +
+				"Can also be specified with KTAPI_RETRY_MAX_ATTEMPTS environment variable.",
+		},
+		minDelayKey: {
+			Type:        schema.TypeString,
+			Optional:    true,
+			DefaultFunc: schema.EnvDefaultFunc("KTAPI_RETRY_MIN_DELAY", nil),
+			Description: "Minimum delay before request retry. " +
+				"Expected Go time duration format, e.g. 1s (see: <https://pkg.go.dev/time#ParseDuration>). " +
```

Comment:
As a user of this provider, I would probably like to just specify seconds here, not GO-specific format

Reply:
Number of seconds/milliseconds is a little bit easier to use, but less flexible.
User might want to specify, e.g. 100ms for MinDelay or 30 minutes for MaxDelay, which is easy to do with Go duration syntax. I'll keep that as it is.